java.lang.AssertionError: assertion failed: position not set (when using incorrect import statement)

## Compiler version

`3.8.3`

## Minimized code
```Scala
import io.circe.parser.parse.decode
```

This tries to import a non-existent `decode` symbol from a method `parse`. The code of `parse` is here for reference:

<details>

```scala
package io.circe

import io.circe.scalajs.convertJsToJson
import scala.scalajs.js.JSON
import scala.util.control.NonFatal

package object parser extends Parser {
  final def parse(input: String): Either[ParsingFailure, Json] = (
    try convertJsToJson(JSON.parse(input))
    catch {
      case NonFatal(exception) => Left(ParsingFailure(exception.getMessage, exception))
    }
  ) match {
    case r @ Right(_)    => r.asInstanceOf[Either[ParsingFailure, Json]]
    case Left(exception) => Left(ParsingFailure(exception.getMessage, exception))
  }
}
```

</details>

## Output (click arrow to expand)

<details>

```scala
java.lang.AssertionError: assertion failed: position not set for Ensuring(null:
  (io.circe.parser.parse :
    (input: String): Either[io.circe.ParsingFailure, io.circe.Json])
) # -1 of class dotty.tools.dotc.ast.Trees$Apply in /tmp/scastie2893000572195522484/src/main/scala/main.scala
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:10)
	at dotty.tools.dotc.typer.Typer$.assertPositioned(Typer.scala:76)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3979)
	at dotty.tools.dotc.typer.Implicits.tryConversion$1(Implicits.scala:1200)
	at dotty.tools.dotc.typer.Implicits.tryConversionForSelection$1(Implicits.scala:1209)
	at dotty.tools.dotc.typer.Implicits.typedImplicit(Implicits.scala:1229)
	at dotty.tools.dotc.typer.Implicits.typedImplicit$(Implicits.scala:880)
	at dotty.tools.dotc.typer.Typer.typedImplicit(Typer.scala:163)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.tryImplicit(Implicits.scala:1307)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.rank$1(Implicits.scala:1477)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicit(Implicits.scala:1647)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicit(Implicits.scala:1771)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.bestImplicit(Implicits.scala:1813)
	at dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:1126)
	at dotty.tools.dotc.typer.Implicits.inferImplicit$(Implicits.scala:880)
	at dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:163)
	at dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:1144)
	at dotty.tools.dotc.typer.Implicits.inferImplicit$(Implicits.scala:880)
	at dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:163)
	at dotty.tools.dotc.typer.Implicits.inferView(Implicits.scala:920)
	at dotty.tools.dotc.typer.Implicits.inferView$(Implicits.scala:880)
	at dotty.tools.dotc.typer.Typer.inferView(Typer.scala:163)
	at dotty.tools.dotc.typer.Implicits.viewExists(Implicits.scala:893)
	at dotty.tools.dotc.typer.Implicits.viewExists$(Implicits.scala:880)
	at dotty.tools.dotc.typer.Typer.viewExists(Typer.scala:163)
	at dotty.tools.dotc.typer.ProtoTypes$Compatibility.isCompatible(ProtoTypes.scala:45)
	at dotty.tools.dotc.typer.ProtoTypes$Compatibility.isCompatible$(ProtoTypes.scala:32)
	at dotty.tools.dotc.typer.Typer.isCompatible(Typer.scala:163)
	at dotty.tools.dotc.typer.ProtoTypes$Compatibility.testCompat$1(ProtoTypes.scala:62)
	at dotty.tools.dotc.typer.ProtoTypes$Compatibility.normalizedCompatible(ProtoTypes.scala:76)
	at dotty.tools.dotc.typer.ProtoTypes$Compatibility.normalizedCompatible$(ProtoTypes.scala:32)
	at dotty.tools.dotc.typer.Typer.normalizedCompatible(Typer.scala:163)
	at dotty.tools.dotc.typer.ProtoTypes$SelectionProto.qualifies$1(ProtoTypes.scala:267)
	at dotty.tools.dotc.typer.ProtoTypes$SelectionProto.isMatchedBy$$anonfun$1(ProtoTypes.scala:270)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.hasAltWith(Denotations.scala:649)
	at dotty.tools.dotc.typer.ProtoTypes$SelectionProto.isMatchedBy(ProtoTypes.scala:270)
	at dotty.tools.dotc.core.TypeComparer.isMatchedByProto(TypeComparer.scala:2349)
	at dotty.tools.dotc.core.TypeComparer.firstTry$1(TypeComparer.scala:356)
	at dotty.tools.dotc.core.TypeComparer.recur(TypeComparer.scala:1629)
	at dotty.tools.dotc.core.TypeComparer.isSubType(TypeComparer.scala:226)
	at dotty.tools.dotc.core.TypeComparer.isSubType(TypeComparer.scala:236)
	at dotty.tools.dotc.core.TypeComparer.topLevelSubType(TypeComparer.scala:144)
	at dotty.tools.dotc.core.TypeComparer.testSubType(TypeComparer.scala:161)
	at dotty.tools.dotc.core.TypeComparer$.testSubType(TypeComparer.scala:3479)
	at dotty.tools.dotc.typer.Typer.adaptNoArgsOther$1(Typer.scala:4810)
	at dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:4897)
	at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:5166)
	at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:4366)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3984)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3989)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:4101)
	at dotty.tools.dotc.typer.Typer.typeSelectOnTerm$1(Typer.scala:1076)
	at dotty.tools.dotc.typer.Typer.typedSelect(Typer.scala:1130)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3778)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3896)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3984)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3989)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:4101)
	at dotty.tools.dotc.typer.Namer.typedAheadExpr$$anonfun$1(Namer.scala:1831)
	at dotty.tools.dotc.typer.Namer.typedAhead(Namer.scala:1821)
	at dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:1831)
	at dotty.tools.dotc.typer.Namer$Completer.$anonfun$14(Namer.scala:850)
	at dotty.tools.dotc.typer.Typer.typedImportQualifier(Typer.scala:3551)
	at dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:850)
	at dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:1013)
	at dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:876)
	at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:176)
	at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:190)
	at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:192)
	at dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:404)
	at dotty.tools.dotc.typer.Typer.retrieveSym(Typer.scala:3750)
	at dotty.tools.dotc.typer.Typer.typedImport(Typer.scala:3554)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3816)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3897)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3984)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3989)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:4001)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:4058)
	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:3449)
	at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3791)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3795)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3896)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3984)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3989)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:4011)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:4058)
	at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:3584)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3837)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3897)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3984)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3989)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:4101)
	at dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$1(TyperPhase.scala:47)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:533)
	at dotty.tools.dotc.typer.TyperPhase.typeCheck(TyperPhase.scala:53)
	at dotty.tools.dotc.typer.TyperPhase.$anonfun$4(TyperPhase.scala:99)
	at scala.collection.Iterator$$anon$6.hasNext(Iterator.scala:495)
	at scala.collection.Iterator$$anon$9.hasNext(Iterator.scala:597)
	at scala.collection.immutable.List.prependedAll(List.scala:156)
	at scala.collection.immutable.List$.from(List.scala:681)
	at scala.collection.immutable.List$.from(List.scala:681)
	at scala.collection.IterableOps$WithFilter.map(Iterable.scala:906)
	at dotty.tools.dotc.typer.TyperPhase.runOn(TyperPhase.scala:98)
	at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:380)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1324)
	at dotty.tools.dotc.Run.runPhases$1(Run.scala:373)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$2(Run.scala:420)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$adapted$1(Run.scala:420)
	at scala.Function0.apply$mcV$sp(Function0.scala:42)
	at dotty.tools.dotc.Run.showProgress(Run.scala:482)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:420)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:432)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:69)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:432)
	at dotty.tools.dotc.Run.compileSources(Run.scala:319)
	at dotty.tools.dotc.Run.compile(Run.scala:304)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:37)
	at dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:141)
	at dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:196)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:252)
	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:186)
	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:166)
	at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:241)
	at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:166)
	at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:214)
	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:542)
	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:542)
	at sbt.internal.inc.Incremental$.$anonfun$apply$3(Incremental.scala:182)
	at sbt.internal.inc.Incremental$.$anonfun$apply$3$adapted(Incremental.scala:180)
	at sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:458)
	at sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:117)
	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
	at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
	at sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:413)
	at sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:500)
	at sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:400)
	at sbt.internal.inc.Incremental$.apply(Incremental.scala:208)
	at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:542)
	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:496)
	at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
	at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:433)
	at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
	at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2443)
	at sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2393)
	at sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:41)
	at sbt.internal.io.Retry$.apply(Retry.scala:47)
	at sbt.internal.io.Retry$.apply(Retry.scala:29)
	at sbt.internal.io.Retry$.apply(Retry.scala:24)
	at sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:41)
	at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2391)
	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
	at sbt.std.Transform$$anon$4.work(Transform.scala:69)
	at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
	at sbt.Execute.work(Execute.scala:292)
	at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

</details>

## Link to scastie / required dependencies

I've got too little knowledge about where this might originate to try and minimize to something without circe (I've checked `3.3.7` and `3.8.2` - they show the same behavior, so this is no recent regression)

Here's a scastie with the necessary build dependencies added: https://scastie.scala-lang.org/aumann/KLVswMZeQqubO93YRcLVMg

```sbt
libraryDependencies ++= Seq(
  "io.circe" %% "circe-core" % "0.14.15",
  "io.circe" %% "circe-generic" % "0.14.15",
  "io.circe" %% "circe-parser" % "0.14.15"
)
```

--- a/compiler/src/dotty/tools/dotc/typer/Implicits.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Implicits.scala
@@ -889,7 +889,7 @@ trait Implicits:
     && ctx.mode.is(Mode.ImplicitsEnabled)
     && from.isValueType
     && (  from.isValueSubType(to)
-       || inferView(dummyTreeOfType(from), to)
+       || inferView(dummyTreeOfType(from).withSpan(ctx.tree.span), to)
             (using ctx.fresh.addMode(Mode.ImplicitExploration).setExploreTyperState()).isSuccess
           // TODO: investigate why we can't TyperState#test here
        || from.widen.isNamedTupleType && to.derivesFrom(defn.TupleClass)

--- a/tests/neg/i25856.scala
+++ b/tests/neg/i25856.scala
@@ -1,0 +1,5 @@
+// https://github.com/scala/scala3/issues/25856
+object parser:
+  def parse(input: String): Int = ???
+
+import parser.parse.decode // error // error

